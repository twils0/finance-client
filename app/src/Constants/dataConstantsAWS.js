// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
