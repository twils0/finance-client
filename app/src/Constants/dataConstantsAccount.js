export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
