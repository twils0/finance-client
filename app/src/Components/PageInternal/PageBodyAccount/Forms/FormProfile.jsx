import React from 'react';
import PropTypes from 'prop-types';

import { formNames, inputNames } from '../../../../Constants/uiConstantsAccount';

import Form from '../../../Form';
import FlexRow from '../../../Flex/FlexRow';
import Header2 from '../../../Header2';
import Paragraph from '../../../Paragraph';
import Input from '../../../Inputs/Input';

const FormProfile = (props) => {
  const {
    heightRef,
    rowWidthIndent,
    headerWidthBufferRight,
    mediaWidth,
    rowHeight,
    rowHeightBuffer,
    currentFormBool,
    editForm,
    inputs,
    handleInputChange,
  } = props;
  const rowWidthIndentString = `${rowWidthIndent}px`;
  const headerWidthBufferRightString = `${headerWidthBufferRight}px`;
  const mediaWidthString = `${mediaWidth}px`;
  const rowHeightString = `${rowHeight}px`;
  const rowHeightBufferHalfString = `${rowHeightBuffer / 2}px`;
  const formString = formNames.PROFILE;

  return (
    <Form innerRef={heightRef} id={formString} width="100%">
      <FlexRow
        justifyContent="flex-start"
        width="100%"
        minHeight={rowHeightString}
        borderWidth="1px 0"
      >
        <Header2
          padding={`0 ${headerWidthBufferRightString} 0 ${rowWidthIndentString}`}
          width={`calc(35% - ${headerWidthBufferRightString})`}
          mediaWidth={mediaWidthString}
          responsiveWidth={
            !editForm
              ? `calc(30% - ${headerWidthBufferRightString})`
              : `calc(100% - ${rowWidthIndentString})`
          }
          themeFontWeight="light"
        >
          Name:
        </Header2>
        {!editForm ? (
          <Paragraph
            width="65%"
            textAlign="right"
            colorType="text"
            themeFont="text"
            themeFontWeight="normal"
            themeSize="text"
            mediaWidth={mediaWidthString}
            responsiveThemeFont="mobileText"
            responsiveWidth="70%"
          >
            {inputs[inputNames[formString].NAME].value}
          </Paragraph>
        ) : (
          <Input
            id={inputNames[formString].NAME}
            tabIndex={currentFormBool ? '0' : '-1'}
            type="text"
            handleInputChange={handleInputChange}
            value={inputs[inputNames[formString].NAME].value}
            errorMessage={inputs[inputNames[formString].NAME].errorMessage}
            WrapperDiv={{
              width: '65%',
              margin: `${rowHeightBufferHalfString} 0`,
              mediaWidth: mediaWidthString,
              responsiveWidth: `calc(100% - ${rowWidthIndentString})`,
              responsiveMargin: `0 0 ${rowHeightBufferHalfString} 0`,
              responsivePadding: `0 0 0 ${rowWidthIndentString}`,
            }}
            InputStyle={{
              mediaWidth: mediaWidthString,
            }}
          />
        )}
      </FlexRow>
      <FlexRow
        justifyContent="flex-start"
        width="100%"
        minHeight={rowHeightString}
        borderWidth="0 0 1px 0"
      >
        <Header2
          padding={`0 0 0 ${rowWidthIndentString}`}
          width={`calc(35% - ${headerWidthBufferRightString})`}
          mediaWidth={mediaWidthString}
          responsiveWidth={
            !editForm
              ? `calc(30% - ${headerWidthBufferRightString})`
              : `calc(100% - ${rowWidthIndentString})`
          }
          themeFontWeight="light"
        >
          Login Email:
        </Header2>
        {!editForm ? (
          <Paragraph
            width="65%"
            textAlign="right"
            colorType="text"
            themeFont="text"
            themeFontWeight="normal"
            themeSize="text"
            mediaWidth={mediaWidthString}
            responsiveThemeFont="mobileText"
            responsiveWidth="70%"
          >
            {inputs[inputNames[formString].EMAIL].value}
          </Paragraph>
        ) : (
          <Input
            id={inputNames[formString].EMAIL}
            tabIndex={currentFormBool ? '0' : '-1'}
            type="email"
            handleInputChange={handleInputChange}
            value={inputs[inputNames[formString].EMAIL].value}
            errorMessage={inputs[inputNames[formString].EMAIL].errorMessage}
            WrapperDiv={{
              width: '65%',
              margin: `${rowHeightBufferHalfString} 0`,
              mediaWidth: mediaWidthString,
              responsiveWidth: `calc(100% - ${rowWidthIndentString})`,
              responsiveMargin: `0 0 ${rowHeightBufferHalfString} 0`,
              responsivePadding: `0 0 0 ${rowWidthIndentString}`,
            }}
            InputStyle={{
              mediaWidth: mediaWidthString,
            }}
          />
        )}
      </FlexRow>
      <FlexRow
        justifyContent="flex-start"
        width="100%"
        minHeight={rowHeightString}
        borderWidth="0 0 1px 0"
      >
        <Header2
          padding={`0 0 0 ${rowWidthIndentString}`}
          width={`calc(35% - ${headerWidthBufferRightString})`}
          mediaWidth={mediaWidthString}
          responsiveWidth={
            !editForm
              ? `calc(30% - ${headerWidthBufferRightString})`
              : `calc(100% - ${rowWidthIndentString})`
          }
          themeFontWeight="light"
        >
          Additional Email:
        </Header2>
        {!editForm ? (
          <Paragraph
            width="65%"
            textAlign="right"
            colorType="text"
            themeFont="text"
            themeFontWeight="normal"
            themeSize="text"
            mediaWidth={mediaWidthString}
            responsiveThemeFont="mobileText"
            responsiveWidth="70%"
          >
            {inputs[inputNames[formString].EMAIL_ADDITIONAL].value}
          </Paragraph>
        ) : (
          <Input
            id={inputNames[formString].EMAIL_ADDITIONAL}
            tabIndex={currentFormBool ? '0' : '-1'}
            type="email"
            handleInputChange={handleInputChange}
            value={inputs[inputNames[formString].EMAIL_ADDITIONAL].value}
            errorMessage={inputs[inputNames[formString].EMAIL_ADDITIONAL].errorMessage}
            WrapperDiv={{
              width: '65%',
              margin: `${rowHeightBufferHalfString} 0`,
              mediaWidth: mediaWidthString,
              responsiveWidth: `calc(100% - ${rowWidthIndentString})`,
              responsiveMargin: `0 0 ${rowHeightBufferHalfString} 0`,
              responsivePadding: `0 0 0 ${rowWidthIndentString}`,
            }}
            InputStyle={{
              mediaWidth: mediaWidthString,
            }}
          />
        )}
      </FlexRow>
      <FlexRow
        justifyContent="flex-start"
        width="100%"
        minHeight={rowHeightString}
        borderWidth="0 0 1px 0"
      >
        <Header2
          padding={`0 0 0 ${rowWidthIndentString}`}
          width={`calc(35% - ${headerWidthBufferRightString})`}
          mediaWidth={mediaWidthString}
          responsiveWidth={
            !editForm
              ? `calc(35% - ${headerWidthBufferRightString})`
              : `calc(100% - ${rowWidthIndentString})`
          }
          themeFontWeight="light"
        >
          Mobile Phone:
        </Header2>
        {!editForm ? (
          <Paragraph
            width="65%"
            textAlign="right"
            colorType="text"
            themeFont="text"
            themeFontWeight="normal"
            themeSize="text"
            mediaWidth={mediaWidthString}
            responsiveThemeFont="mobileText"
          >
            {inputs[inputNames[formString].PHONE].value}
          </Paragraph>
        ) : (
          <Input
            id={inputNames[formString].PHONE}
            tabIndex={currentFormBool ? '0' : '-1'}
            type="tel"
            handleInputChange={handleInputChange}
            value={inputs[inputNames[formString].PHONE].value}
            errorMessage={inputs[inputNames[formString].PHONE].errorMessage}
            WrapperDiv={{
              width: '65%',
              margin: `${rowHeightBufferHalfString} 0`,
              mediaWidth: mediaWidthString,
              responsiveWidth: `calc(100% - ${rowWidthIndentString})`,
              responsiveMargin: `0 0 ${rowHeightBufferHalfString} 0`,
              responsivePadding: `0 0 0 ${rowWidthIndentString}`,
            }}
            InputStyle={{
              mediaWidth: mediaWidthString,
            }}
          />
        )}
      </FlexRow>
    </Form>
  );
};

FormProfile.propTypes = {
  heightRef: PropTypes.func.isRequired,
  rowWidthIndent: PropTypes.number.isRequired,
  headerWidthBufferRight: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  rowHeight: PropTypes.number.isRequired,
  rowHeightBuffer: PropTypes.number.isRequired,
  currentFormBool: PropTypes.bool.isRequired,
  editForm: PropTypes.bool.isRequired,
  inputs: PropTypes.object.isRequired,
  handleInputChange: PropTypes.func.isRequired,
};

FormProfile.defaultProps = {};

export default FormProfile;
