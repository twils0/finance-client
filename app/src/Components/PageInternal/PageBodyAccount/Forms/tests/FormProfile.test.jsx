import React from 'react';
import { shallow } from 'enzyme';

import { initialState as initialStateUIAccount } from '../../../../../Reducers/uiReducersAccount';
import { formNames, inputNames } from '../../../../../Constants/uiConstantsAccount';

import FormProfile from '../FormProfile';

const heightRef = jest.fn();
const handleInputChange = jest.fn();

const baseProps = {
  heightRef,
  rowWidthIndent: 10,
  headerWidthBufferRight: 11,
  mediaWidth: 12,
  rowHeight: 13,
  rowHeightBuffer: 14,
  currentFormBool: false,
  editForm: false,
  inputs: null,
  handleInputChange,
};

const name = 'testName';
const email = 'testEmail';
const emailAdditional = 'testEmailAdditional';
const phone = '239-555-0000';

const shallowComponent = props => shallow(<FormProfile {...props} />);

describe('Components', () => {
  describe('PageInternal', () => {
    describe('PageBodyAccount', () => {
      describe('Forms', () => {
        describe('FormProfile', () => {
          afterEach(() => {
            heightRef.mockReset();
            handleInputChange.mockReset();
          });

          it('shallow renders correctly, when currentFormBool and editForm is false', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { inputs } = stateBeforeUIAccount.forms[formNames.PROFILE];

            inputs[inputNames[formNames.PROFILE].NAME].value = name;
            inputs[inputNames[formNames.PROFILE].EMAIL].value = email;
            inputs[inputNames[formNames.PROFILE].EMAIL_ADDITIONAL].value = emailAdditional;
            inputs[inputNames[formNames.PROFILE].PHONE].value = phone;

            const props = { ...baseProps, inputs };

            const wrapper = shallowComponent(props);

            expect(wrapper).toMatchSnapshot();
          });

          it('shallow renders correctly, when currentFormBool and editForm is true', async () => {
            const stateBeforeUIAccount = JSON.parse(JSON.stringify(initialStateUIAccount));
            const { inputs } = stateBeforeUIAccount.forms[formNames.PROFILE];

            inputs[inputNames[formNames.PROFILE].NAME].value = name;
            inputs[inputNames[formNames.PROFILE].EMAIL].value = email;
            inputs[inputNames[formNames.PROFILE].EMAIL_ADDITIONAL].value = emailAdditional;
            inputs[inputNames[formNames.PROFILE].PHONE].value = phone;

            const props = {
              ...baseProps,
              inputs,
              currentFormBool: true,
              editForm: true,
            };

            const wrapper = shallowComponent(props);

            expect(wrapper).toMatchSnapshot();
          });
        });
      });
    });
  });
});
