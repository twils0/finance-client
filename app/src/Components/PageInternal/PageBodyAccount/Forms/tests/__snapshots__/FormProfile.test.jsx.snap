// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components PageInternal PageBodyAccount Forms FormProfile shallow renders correctly, when currentFormBool and editForm is false 1`] = `
<Form__FormStyle
  alignItems={null}
  alignSelf={null}
  background={null}
  borderColorType="border"
  borderRadius={null}
  borderStyle={null}
  borderWidth={null}
  boxSizing={null}
  color={Object {}}
  display={null}
  flex={null}
  flexDirection={null}
  height={null}
  id="profileForm"
  justifyContent={null}
  margin={null}
  mediaWidth={null}
  padding={null}
  position={null}
  themeColor="primary"
  width="100%"
>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 11px 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(30% - 11px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Name:
    </Header2>
    <Paragraph__P
      alignItems={null}
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="text"
      display={null}
      flex={null}
      flexDirection={null}
      flexWrap={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      justifyContent={null}
      letterSpacing={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      minWidth={null}
      padding={null}
      responsiveThemeFont="mobileText"
      responsiveWidth="70%"
      textAlign="right"
      themeColor="primary"
      themeFont="text"
      themeFontWeight="normal"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="text"
      type="text"
      whiteSpace={null}
      width="65%"
    >
      testName
    </Paragraph__P>
  </FlexRow>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="0 0 1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 0 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(30% - 11px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Login Email:
    </Header2>
    <Paragraph__P
      alignItems={null}
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="text"
      display={null}
      flex={null}
      flexDirection={null}
      flexWrap={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      justifyContent={null}
      letterSpacing={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      minWidth={null}
      padding={null}
      responsiveThemeFont="mobileText"
      responsiveWidth="70%"
      textAlign="right"
      themeColor="primary"
      themeFont="text"
      themeFontWeight="normal"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="text"
      type="text"
      whiteSpace={null}
      width="65%"
    >
      testEmail
    </Paragraph__P>
  </FlexRow>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="0 0 1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 0 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(30% - 11px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Additional Email:
    </Header2>
    <Paragraph__P
      alignItems={null}
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="text"
      display={null}
      flex={null}
      flexDirection={null}
      flexWrap={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      justifyContent={null}
      letterSpacing={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      minWidth={null}
      padding={null}
      responsiveThemeFont="mobileText"
      responsiveWidth="70%"
      textAlign="right"
      themeColor="primary"
      themeFont="text"
      themeFontWeight="normal"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="text"
      type="text"
      whiteSpace={null}
      width="65%"
    >
      testEmailAdditional
    </Paragraph__P>
  </FlexRow>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="0 0 1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 0 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(35% - 11px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Mobile Phone:
    </Header2>
    <Paragraph__P
      alignItems={null}
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="text"
      display={null}
      flex={null}
      flexDirection={null}
      flexWrap={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      justifyContent={null}
      letterSpacing={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      minWidth={null}
      padding={null}
      responsiveThemeFont="mobileText"
      responsiveWidth={null}
      textAlign="right"
      themeColor="primary"
      themeFont="text"
      themeFontWeight="normal"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="text"
      type="text"
      whiteSpace={null}
      width="65%"
    >
      239-555-0000
    </Paragraph__P>
  </FlexRow>
</Form__FormStyle>
`;

exports[`Components PageInternal PageBodyAccount Forms FormProfile shallow renders correctly, when currentFormBool and editForm is true 1`] = `
<Form__FormStyle
  alignItems={null}
  alignSelf={null}
  background={null}
  borderColorType="border"
  borderRadius={null}
  borderStyle={null}
  borderWidth={null}
  boxSizing={null}
  color={Object {}}
  display={null}
  flex={null}
  flexDirection={null}
  height={null}
  id="profileForm"
  justifyContent={null}
  margin={null}
  mediaWidth={null}
  padding={null}
  position={null}
  themeColor="primary"
  width="100%"
>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 11px 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(100% - 10px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Name:
    </Header2>
    <Input
      Error={Object {}}
      InputStyle={
        Object {
          "mediaWidth": "12px",
        }
      }
      Label={Object {}}
      WrapperDiv={
        Object {
          "margin": "7px 0",
          "mediaWidth": "12px",
          "responsiveMargin": "0 0 7px 0",
          "responsivePadding": "0 0 0 10px",
          "responsiveWidth": "calc(100% - 10px)",
          "width": "65%",
        }
      }
      dataSet={Object {}}
      errorMessage=""
      handleInputChange={[MockFunction]}
      id="profileName"
      label={null}
      tabIndex="0"
      type="text"
      value="testName"
    />
  </FlexRow>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="0 0 1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 0 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(100% - 10px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Login Email:
    </Header2>
    <Input
      Error={Object {}}
      InputStyle={
        Object {
          "mediaWidth": "12px",
        }
      }
      Label={Object {}}
      WrapperDiv={
        Object {
          "margin": "7px 0",
          "mediaWidth": "12px",
          "responsiveMargin": "0 0 7px 0",
          "responsivePadding": "0 0 0 10px",
          "responsiveWidth": "calc(100% - 10px)",
          "width": "65%",
        }
      }
      dataSet={Object {}}
      errorMessage=""
      handleInputChange={[MockFunction]}
      id="profileEmail"
      label={null}
      tabIndex="0"
      type="email"
      value="testEmail"
    />
  </FlexRow>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="0 0 1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 0 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(100% - 10px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Additional Email:
    </Header2>
    <Input
      Error={Object {}}
      InputStyle={
        Object {
          "mediaWidth": "12px",
        }
      }
      Label={Object {}}
      WrapperDiv={
        Object {
          "margin": "7px 0",
          "mediaWidth": "12px",
          "responsiveMargin": "0 0 7px 0",
          "responsivePadding": "0 0 0 10px",
          "responsiveWidth": "calc(100% - 10px)",
          "width": "65%",
        }
      }
      dataSet={Object {}}
      errorMessage=""
      handleInputChange={[MockFunction]}
      id="profileEmailAdditional"
      label={null}
      tabIndex="0"
      type="email"
      value="testEmailAdditional"
    />
  </FlexRow>
  <FlexRow
    alignItems={null}
    alignSelf={null}
    borderColorType="border"
    borderRadius={null}
    borderStyle={null}
    borderWidth="0 0 1px 0"
    boxSizing={null}
    color={Object {}}
    display={null}
    flex={null}
    flexWrap={null}
    height={null}
    justifyContent="flex-start"
    left={null}
    margin={null}
    mediaWidth={null}
    minHeight="13px"
    minWidth={null}
    overflowX={null}
    overflowY={null}
    padding={null}
    position={null}
    responsiveAlignSecurities={null}
    responsiveJustifyContent={null}
    responsiveMargin={null}
    responsivePadding={null}
    responsivePosition={null}
    themeColor="primary"
    top={null}
    width="100%"
  >
    <Header2
      alignSelf={null}
      background={null}
      borderColorType="border"
      borderRadius={null}
      borderStyle={null}
      borderWidth={null}
      boxSizing={null}
      color={Object {}}
      colorType="default"
      display={null}
      fontFamily={null}
      fontSize={null}
      fontWeight={null}
      height={null}
      id={null}
      lineHeight={null}
      margin={null}
      mediaWidth="12px"
      padding="0 0 0 10px"
      position={null}
      responsiveJustifyContent={null}
      responsiveMargin={null}
      responsivePadding={null}
      responsiveTextAlign={null}
      responsiveWidth="calc(100% - 10px)"
      textAlign={null}
      themeColor="primary"
      themeFont="default"
      themeFontWeight="light"
      themeLetterSpacing="default"
      themeLineHeight="default"
      themeSize="h2"
      type="text"
      whiteSpace={null}
      width="calc(35% - 11px)"
    >
      Mobile Phone:
    </Header2>
    <Input
      Error={Object {}}
      InputStyle={
        Object {
          "mediaWidth": "12px",
        }
      }
      Label={Object {}}
      WrapperDiv={
        Object {
          "margin": "7px 0",
          "mediaWidth": "12px",
          "responsiveMargin": "0 0 7px 0",
          "responsivePadding": "0 0 0 10px",
          "responsiveWidth": "calc(100% - 10px)",
          "width": "65%",
        }
      }
      dataSet={Object {}}
      errorMessage=""
      handleInputChange={[MockFunction]}
      id="profilePhone"
      label={null}
      tabIndex="0"
      type="tel"
      value="239-555-0000"
    />
  </FlexRow>
</Form__FormStyle>
`;
